From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Axtens <dja@axtens.net>
Date: Mon, 6 Sep 2021 15:46:11 +1000
Subject: [PATCH] powerpc: Drop Open Hack'Ware

Open Hack'Ware was an alternative firmware of powerpc under QEMU.

The last commit to any Open Hack'Ware repo I can find is from 2014 [1].

Open Hack'Ware was used for the QEMU "prep" machine type, which was
deprecated in QEMU in commit 54c86f5a4844 (hw/ppc: deprecate the
machine type 'prep', replaced by '40p') in QEMU v3.1, and had reportedly
been broken for years before without anyone noticing. Support was removed
in February 2020 by commit b2ce76a0730e (hw/ppc/prep: Remove the
deprecated "prep" machine and the OpenHackware BIOS).

Open Hack'Ware's limitations require some messy code in GRUB. This
complexity is not worth carrying any more.

Remove detection of Open Hack'Ware. We will clean up the feature flags
in following commits.

[1]: https://github.com/qemu/openhackware and
     https://repo.or.cz/w/openhackware.git are QEMU submodules. They have
     only small changes on top of OHW v0.4.1, which was imported into
     QEMU SCM in 2010. I can't find anything resembling an official repo
     any more.

Signed-off-by: Daniel Axtens <dja@axtens.net>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
(cherry picked from commit f9ce538eec88c5cffbfde021c4e8a95a5e9d0e8f)
---
 grub-core/kern/ieee1275/cmain.c | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/grub-core/kern/ieee1275/cmain.c b/grub-core/kern/ieee1275/cmain.c
index dce7b84922..cb42f60ebe 100644
--- a/grub-core/kern/ieee1275/cmain.c
+++ b/grub-core/kern/ieee1275/cmain.c
@@ -49,7 +49,6 @@ grub_ieee1275_find_options (void)
   grub_ieee1275_phandle_t root;
   grub_ieee1275_phandle_t options;
   grub_ieee1275_phandle_t openprom;
-  grub_ieee1275_phandle_t bootrom;
   int rc;
   grub_uint32_t realmode = 0;
   char tmp[256];
@@ -198,21 +197,6 @@ grub_ieee1275_find_options (void)
 
       grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_HAS_CURSORONOFF);
     }
-
-  if (! grub_ieee1275_finddevice ("/rom/boot-rom", &bootrom)
-      || ! grub_ieee1275_finddevice ("/boot-rom", &bootrom))
-    {
-      rc = grub_ieee1275_get_property (bootrom, "model", tmp, sizeof (tmp), 0);
-      if (rc >= 0 && !grub_strncmp (tmp, "PPC Open Hack'Ware",
-				    sizeof ("PPC Open Hack'Ware") - 1))
-	{
-	  grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_BROKEN_OUTPUT);
-	  grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_CANNOT_SET_COLORS);
-	  grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_CANNOT_INTERPRET);
-	  grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_FORCE_CLAIM);
-	  grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_NO_ANSI);
-	}
-    }
 }
 
 void
